overlord/snapstate/aliasesv2.go

```diff
@@ -378,3 +403,80 @@ func pruneAutoAliases(st *state.State, curAliases map[string]*AliasTarget, autoA
 	}
 	return newAliases
 }
+
+// transition to aliases v2
+func (m *SnapManager) ensureAliasesV2() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var aliasesV1 map[string]interface{}
+	err := m.state.Get("aliases", &aliasesV1)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if len(aliasesV1) == 0 {
+		m.state.Set("aliases", nil)
```

Comment:
Will this trigger the "was written to" flag of the state, or are we smart about this already?  We need to make sure it doesn't flag as written, otherwise every single ensure will hit the disk.